File: java/src/main/java/com/android/widget/RecyclerView/AATest/TestRecyclerViewActivity.java
package com.android.widget.RecyclerView.AATest;

import android.os.Bundle;

import com.android.frame.mvc.BaseActivity;
import com.android.java.databinding.ActivityCommonLayoutBinding;
import com.android.util.CommonLayoutUtil;

/**
 * Create by xuzhb on 2020/1/21
 * Desc:
 */
public class TestRecyclerViewActivity extends BaseActivity<ActivityCommonLayoutBinding> {

    @Override
    public void handleView(Bundle savedInstanceState) {
        CommonLayoutUtil.initCommonLayout(
                this, "RecyclerView",
                "单一布局", "多布局", "头部Header", "尾部Footer", "下拉刷新和上拉加载更多",
                "单一布局上拉加载更多", "多布局上拉加载更多"
        );
    }

    @Override
    public void initListener() {
        binding.btn1.setOnClickListener(v -> {  //单一布局
            startActivity(TestSingleAdapterActivity.class);
        });
        binding.btn2.setOnClickListener(v -> {  //多布局
            startActivity(TestMultiAdapterActivity.class);
        });
        binding.btn3.setOnClickListener(v -> {  //头部Header
            startActivity(TestHeaderAdapterActivity.class);
        });
        binding.btn4.setOnClickListener(v -> {  //尾部Footer
            startActivity(TestFooterAdapterActivity.class);
        });
        binding.btn5.setOnClickListener(v -> {  //下拉刷新和上拉加载更多
            startActivity(TestLoadMoreWrapperActivity.class);
        });
        binding.btn6.setOnClickListener(v -> {  //单一布局上拉加载更多
            startActivity(TestSingleLoadMoreAdapterActivity.class);
        });
        binding.btn7.setOnClickListener(v -> {  //多布局上拉加载更多
            startActivity(TestMultiLoadMoreAdapterActivity.class);
        });
    }

    @Override
    public ActivityCommonLayoutBinding getViewBinding() {
        return ActivityCommonLayoutBinding.inflate(getLayoutInflater());
    }

}

File: kotlin/src/main/java/com/android/widget/RecyclerView/AATest/TestRecyclerViewActivity.kt
package com.android.widget.RecyclerView.AATest

import android.os.Bundle
import com.android.basicproject.R
import com.android.frame.mvc.BaseActivity
import com.android.util.initCommonLayout
import kotlinx.android.synthetic.main.activity_common_layout.*

/**
 * Created by xuzhb on 2019/10/30
 * Desc:
 */
class TestRecyclerViewActivity : BaseActivity() {

    override fun handleView(savedInstanceState: Bundle?) {
        initCommonLayout(
            this, "RecyclerView",
            "单一布局", "多布局", "头部Header", "尾部Footer", "下拉刷新和上拉加载更多",
            "单一布局上拉加载更多", "多布局上拉加载更多"
        )
    }

    override fun initListener() {
        btn1.setOnClickListener {
            startActivity(TestSingleAdapterActivity::class.java)
        }
        btn2.setOnClickListener {
            startActivity(TestMultiAdapterActivity::class.java)
        }
        btn3.setOnClickListener {
            startActivity(TestHeaderAdapterActivity::class.java)
        }
        btn4.setOnClickListener {
            startActivity(TestFooterAdapterActivity::class.java)
        }
        btn5.setOnClickListener {
            startActivity(TestLoadMoreWrapperActivity::class.java)
        }
        btn6.setOnClickListener {
            startActivity(TestSingleLoadMoreAdapterActivity::class.java)
        }
        btn7.setOnClickListener {
            startActivity(TestMultiLoadMoreAdapterActivity::class.java)
        }
    }

    override fun getLayoutId(): Int = R.layout.activity_common_layout
}